### PR TITLE
Show tournament supporter banner in profile page

### LINKED ISSUE #2535: World Cup supporter banners is invisible on new site
image
this banner :v

### PR DESCRIPTION
**Pull request by nanaya**
Fixes #2535. Not sure if linking to tournament page is useful.

Copypasta of required `.env` variables:

```bash
TOURNAMENT_BANNER_CURRENT_ID=
# Image path/url for tournament supporter banner.
# Will be appended with country abbreviation and `.jpg`. `@2x` version is also required (prepended before `.jpg`).
# Example:
#   prefix: https://assets.ppy.sh/tournament-banners/official/twc2018_banner_
#   image path: https://assets.ppy.sh/tournament-banners/official/twc2018_banner_JP.jpg
#   image path @2x: https://assets.ppy.sh/tournament-banners/official/twc2018_banner_JP@2x.jpg
TOURNAMENT_BANNER_CURRENT_PREFIX=

TOURNAMENT_BANNER_PREVIOUS_ID=
# Same as `_CURRENT_PREFIX`.
TOURNAMENT_BANNER_PREVIOUS_PREFIX=
# Winning country code
TOURNAMENT_BANNER_PREVIOUS_WINNER_ID=
```

### CODE REVIEW

File: app/Models/ProfileBanner.php [hunk 1]
```diff
@@ -42,4 +43,44 @@ public function country()
     {
         return $this->belongsTo(Country::class, 'country_acronym');
     }
+
+    public function macroVisible()
```
Comment (nekodex) [inline app/Models/ProfileBanner.php]: Maybe call this `current` or something instead of `visible`? Kinda ambiguous.

File: config/osu.php [hunk 2]
```diff
@@ -93,6 +93,17 @@
         'notice' => presence(str_replace('\n', "\n", env('STORE_NOTICE'))),
     ],
     'twitch_client_id' => env('TWITCH_CLIENT_ID'),
+    'tournament_support' => [
```
Comment (nekodex) [inline config/osu.php]: Hrm, maybe `tournament_banner(s)` for these variables makes more sense than `tournament_support`?

File: app/Transformers/UserTransformer.php [hunk 3]
```diff
@@ -43,6 +43,7 @@ class UserTransformer extends Fractal\TransformerAbstract
         'statistics',
         'unranked_beatmapset_count',
         'user_achievements',
+        'visible_banner',
```
Comment (nekodex) [inline app/Transformers/UserTransformer.php]: `tournament_banner`, `tournament_support_banner` or something instead? `visible_banner` feels way too ambiguous at the `UserTransformer` level

File: resources/assets/coffee/react/profile-page/header-main.coffee [hunk 4]
```diff
@@ -73,23 +73,34 @@ class ProfilePage.HeaderMain extends React.Component
...
       div
         className: 'profile-header__column profile-header__column--info'
-        el ProfilePage.Badges, badges: @props.user.badges
-        el ProfilePage.HeaderInfo, user: @props.user, currentMode: @props.currentMode
+
+        div
+          className: 'profile-header__info profile-header__info--top'
+          if @props.user.visible_banner.id?
+            a
+              href: laroute.route('tournaments.show', tournament: @props.user.visible_banner.tournament_id)
```
Comment (nekodex) [inline resources/assets/coffee/react/profile-page/header-main.coffee]: I wonder if linking to that page is even useful in its current state...

File: resources/assets/coffee/react/profile-page/header-main.coffee [hunk 5]
```diff
@@ -73,23 +73,34 @@ class ProfilePage.HeaderMain extends React.Component
...
+              el Img2x,
+                src: @props.user.visible_banner.image
+                className: 'profile-header__tournament-support'
```
Comment (nekodex) [inline resources/assets/coffee/react/profile-page/header-main.coffee]: See other comment about `tournament_support` naming.

File: app/Models/ProfileBanner.php (hunk 1 above)
Comment (nanaya) [reply app/Models/ProfileBanner.php]: Went with `active` because it might be for either current or previous tournament.

**Pull request merged**